strawberry/schema/schema_converter.py

```diff
@@ -308,8 +311,91 @@ def get_graphql_fields() -> Dict[str, GraphQLField]:
 
         return graphql_object_type
 
+    def _get_arguments(
+        self,
+        field: StrawberryField,
+        source: Any,
+        info: Any,
+        kwargs: Dict[str, Any],
+    ) -> Tuple[List[Any], Dict[str, Any]]:
+        if field.base_resolver is None:
+            return [], {}
+
+        kwargs = convert_arguments(kwargs, field.arguments)
+
+        # the following code allows to omit info and root arguments
+        # by inspecting the original resolver arguments,
+        # if it asks for self, the source will be passed as first argument
+        # if it asks for root, the source it will be passed as kwarg
+        # if it asks for info, the info will be passed as kwarg
+
+        args = []
+
+        if field.base_resolver.has_self_arg:
+            args.append(source)
+
+        if field.base_resolver.has_root_arg:
+            kwargs["root"] = source
+
+        if field.base_resolver.has_info_arg:
+            kwargs["info"] = info
+
+        return args, kwargs
```

Comment:
Can this methods be embedded in `from_resolver` with the other functions? It doesn't need any state on the class and is only consumed by that one method.

Reply:
I know that `from_resolver` is getting quite long, but I still think that would be cleaner

Reply:
yes, can do :)